File: devel/libjob_queue/src/workflow_job_monitor.c
/*
   Copyright (C) 2014  Statoil ASA, Norway.
    
   The file 'workflow_job_monitor.c' is part of ERT - Ensemble based Reservoir Tool.
    
   ERT is free software: you can redistribute it and/or modify 
   it under the terms of the GNU General Public License as published by 
   the Free Software Foundation, either version 3 of the License, or 
   (at your option) any later version. 
    
   ERT is distributed in the hope that it will be useful, but WITHOUT ANY 
   WARRANTY; without even the implied warranty of MERCHANTABILITY or 
   FITNESS FOR A PARTICULAR PURPOSE.   
    
   See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html> 
   for more details. 
*/

#include <ert/job_queue/workflow_job_monitor.h>
#include <ert/util/util.h>

struct workflow_job_monitor_struct{
    pid_t   processid;
    bool    blocking;
};

workflow_job_monitor_type * workflow_job_monitor_alloc(){
    workflow_job_monitor_type * monitor = util_malloc( sizeof * monitor);
    monitor->blocking = true;
    return monitor;
}

void workflow_job_monitor_free(workflow_job_monitor_type *monitor){
    free(monitor);
}

void workflow_job_monitor_set_pid(workflow_job_monitor_type * monitor, pid_t pid){
    monitor->processid = pid;
}

pid_t workflow_job_monitor_get_pid(workflow_job_monitor_type * monitor){
    return monitor->processid;
}

bool workflow_job_monitor_get_blocking(workflow_job_monitor_type * monitor){
    return monitor->blocking;
}

void workflow_job_monitor_set_blocking(workflow_job_monitor_type * monitor, bool blocking){
    monitor->blocking = blocking;
}

File: devel/python/python/ert/job_queue/workflow_job_monitor.py
from ert.cwrap import BaseCClass, CWrapper
from ert.job_queue import JOB_QUEUE_LIB


class WorkflowJobMonitor(BaseCClass):

    def __init__(self):
        c_ptr = WorkflowJobMonitor.cNamespace().alloc()
        super(WorkflowJobMonitor, self).__init__(c_ptr)

    def setPID(self, pid):
        """
        @type pid: int
        """
        WorkflowJobMonitor.cNamespace().set_pid(self, pid)

    def getPID(self):
        """ @rtype: int """
        return WorkflowJobMonitor.cNamespace().get_pid(self)

    def setBlocking(self, blocking):
        """
        @type blocking: bool
        """
        WorkflowJobMonitor.cNamespace().set_blocking(self, blocking)

    def isBlocking(self):
        """ @rtype: bool """
        return WorkflowJobMonitor.cNamespace().is_blocking(self)

    def free(self):
        WorkflowJobMonitor.cNamespace().free(self)


CWrapper.registerObjectType("workflow_job_monitor", WorkflowJobMonitor)

cwrapper = CWrapper(JOB_QUEUE_LIB)

WorkflowJobMonitor.cNamespace().alloc    = cwrapper.prototype("c_void_p workflow_job_monitor_alloc()")
WorkflowJobMonitor.cNamespace().free     = cwrapper.prototype("void     workflow_job_monitor_free(workflow_job_monitor)")

WorkflowJobMonitor.cNamespace().set_pid = cwrapper.prototype("void workflow_job_monitor_set_pid(workflow_job_monitor, int)")
WorkflowJobMonitor.cNamespace().get_pid = cwrapper.prototype("int workflow_job_monitor_get_pid(workflow_job_monitor)")
WorkflowJobMonitor.cNamespace().set_blocking = cwrapper.prototype("void workflow_job_monitor_set_blocking(workflow_job_monitor, bool)")
WorkflowJobMonitor.cNamespace().is_blocking = cwrapper.prototype("bool workflow_job_monitor_get_blocking(workflow_job_monitor)")
